crates/ruma-server-util/Cargo.toml

```diff
@@ -17,9 +17,11 @@ rustdoc-args = ["--cfg", "docsrs"]
 
 [dependencies]
 headers = "0.4.0"
+http = { workspace = true }
+http-auth = { version = "0.1.9", default-features = false }
```

Comment:
Seems that the spec currently recommends servers to have behavior that is invalid under RFC7230:

https://spec.matrix.org/unstable/server-server-api/
> For compatibility with older servers, the recipient should allow colons to be included in values without requiring the value to be enclosed in quotes.

https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.6
> Most HTTP header field values are defined using common syntax
   components (token, quoted-string, and comment) separated by
   whitespace or specific delimiting characters.  Delimiters are chosen
   from the set of US-ASCII visual characters not allowed in a token
   (DQUOTE and "(),/:;<=>?@[\]{}").

Reply:
That's annoying because that means that we cannot use a RFC7230-compliant implementation after all. However it is only a _should_, so we could ignore it.

Our 2 choices are then:
1. Use `http-auth` and only allow headers that are fully compliant with the RFC.
2. Implement our own parser and follow the quirks of the spec.

Given that I am lazy, I would prefer the first choice. But if you think that it will be problematic to not be fully backwards-compatible, we need to use the second choice.

I don't know what versions of servers used the unquoted values, but the quotes have been clarified in the spec 2 years ago.

Reply:
It's probably fine, since this should only really effect the key and signature, which should probably have already been quoted (I am not aware of servers that weren't already doing this).